schemachanger: enable adding/dropping path of constraint name

Xiang-Gu opened the pull request:
Enable adding/dropping path of constraint name. It's considered
a simple dependent of constraint, so it transition directly between absent
and public.

Fixes #89665
Release note: None

Xiang-Gu committed:
schemachanger: enable adding/dropping path of constraint_name

ConstraintName element is deemed simple dependents of constraint and
hence has transition between PUBLIC <==> ABSENT. This commit thus
is primarily implementing the scop `SetConstraintName`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/90840)
<!-- Reviewable:end -->

Xiang-Gu committed:
scmutationexec: retain check constraint in the dropping path on `Checks`

The old schema changer has the behavior of retaining the to-be-dropped
check constraint on the `Checks` slice. We will do the same in the
new schema changer.

Xiang-Gu commented:
closing as it's now part of https://github.com/cockroachdb/cockroach/pull/91153
